utils/Felt.go

```diff
@@ -87,3 +90,58 @@ func FeltArrToBigIntArr(f []*felt.Felt) []*big.Int {
 	}
 	return bigArr
 }
+
+// StringToByteArrFelt converts string to array of Felt objects.
+// The returned array of felts will be of the format
+//
+// [number of felts with 31 characters in length, 31 byte felts..., pending word with max size of 30 bytes, pending words bytes size]
+//
+// For further explanation, refer the [article]
+//
+// Parameters:
+//
+// - s: string/bytearray to convert
+//
+// Returns:
+//
+// - []*felt.Felt: the array of felt.Felt objects
+//
+// - error: an error, if any
+//
+// [article]: https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/#serialization_of_byte_arrays
+func StringToByteArrFelt(s string) ([]*felt.Felt, error) {
+	const SHORT_LENGTH = 31
+	exp := fmt.Sprintf(".{1,%d}", SHORT_LENGTH)
+	r := regexp.MustCompile(exp)
+
+	arr := r.FindAllString(s, -1)
+	if len(arr) == 0 {
+		return []*felt.Felt{}, fmt.Errorf("invalid string no matches found, s: %s", s)
+	}
+
+	hexarr := []string{}
+	var (
+		count uint64
+		size  uint64
+	)
+
+	for _, val := range arr {
+		if len(val) == SHORT_LENGTH {
+			count += 1
+		} else {
+			size = uint64(len(val))
+		}
+		hexarr = append(hexarr, hex.EncodeToString([]byte(val)))
+	}
+
+	harr, err := HexArrToFelt(hexarr)
+	if err != nil {
+		return nil, err
+	}
+
+	if size > 0 {
```

Comment:
I believe this condition is not necessary. Since the [doc](https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/#serialization_of_byte_arrays) says that this is the "Length of the pending word, in bytes", if the length is 0 we should set it to 0, right? And since 0 is the default uint64 value, we can just set it without any condition.
We can see this with the "Number of 31-byte words in the data array." value in the doc, they push a 0 when there is no 31-byte word, so I believe that should also be the case for pending word length.